File: leetcode/solutions/0003.LongestSubstringWithoutRepeatingCharacters/lengthOfLongestSubstring_test.ts
import { assertEquals } from "https://deno.land/std/testing/asserts.ts";
import lengthOfLongestSubstring from "./lengthOfLongestSubstring.ts";

Deno.test("0003. Longest Substring Without Repeating Characters", () => {
  assertEquals(lengthOfLongestSubstring("abcabcbb"), 3);
  assertEquals(lengthOfLongestSubstring("bbbbb"), 1);
  assertEquals(lengthOfLongestSubstring("pwwkew"), 3);
  assertEquals(lengthOfLongestSubstring("dvdf"), 3);
});

File: leetcode/solutions/0003.LongestSubstringWithoutRepeatingCharacters/lengthOfLongestSubstring_test.go
package solution

import (
	"testing"
)

func Test_0003_LengthOfLongestSubstring(t *testing.T) {
	t.Parallel()

	tests := []struct {
		in  string
		out int
	}{
		{"", 0},
		{"bbbbb", 1},
		{"aab", 2},
		{"dvdf", 3},
		{"abcabcbb", 3},
		{"pwwkew", 3},
		{"abcdefg", 7},
	}

	for _, tt := range tests {
		tt := tt

		t.Run("Test", func(t *testing.T) {
			t.Parallel()

			got := lengthOfLongestSubstring(tt.in)

			if got != tt.out {
				t.Errorf("got %#v want %#v", got, tt.out)
			}
		})
	}
}
